app/Filament/Resources/DocumentResource/RelationManagers/RevisionsRelationManager.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace App\Filament\Resources\DocumentResource\RelationManagers;
+
+use App\Filament\Resources\RevisionResource;
+use App\Models\Document;
+use App\Models\Revision;
+use Filament\Forms;
+use Filament\Forms\Form;
+use Filament\Resources\RelationManagers\RelationManager;
+use Filament\Tables;
+use Filament\Tables\Table;
+use Illuminate\Support\Carbon;
+use Livewire\Features\SupportFileUploads\TemporaryUploadedFile;
+
+class RevisionsRelationManager extends RelationManager
+{
+    protected static string $relationship = 'revisions';
+
+    public function form(Form $form): Form
+    {
+        return $form
+            ->schema([
+                Forms\Components\DatePicker::make('date')
+                    ->label(__('Revision date'))
+                    ->format('Y-m-d')
+                    ->unique()
```

Comment:
So that the record can be updated.
```suggestion
                    ->unique(ignoreRecord: true)
```